tests/integration/synapseclient/integration_test.py

```diff
@@ -123,6 +123,9 @@ def test_entity_version(syn, project, schedule_for_cleanup):
 
     # Delete version 2
     syn.delete(entity, version=2)
+    # Setting versionNumber to None, so we can retrieve the latest version
+    # and confirm it is now version 1.
+    entity.versionNumber = None
```

Comment:
After deleting version 2 of `entity`, this unit test was feeding back the original File Entity that had `versionNumber = 2` in its dictionary of properties.

This used to be fine, and the test used to return the latest version (in this case version 1 now) regardless, because `syn.get` was not getting version number from the `versionNumber` in the File Entity, but rather from the `version` kwarg if it was fed into `syn.get`: see [here for reference ](https://github.com/Sage-Bionetworks/synapsePythonClient/blob/0ea22a414b25ca830e6e2bc5cb181727ccf4cf51/synapseclient/client.py#L1090)

But in this feature branch I updated `syn.get` to pull the version number from either: the `version` kwarg, the `.version` syntax in a synID string, or the `.versionNumber` in an entity, which caused this test to fail. The fix was to update `versionNumber` to None so that the latest version is pulled.

My current stance is that it makes sense for `syn.get` to pull the `.versionNumber` from the Entity object itself, so as to enforce correct Entity objects are fed into `syn.get`, and users are getting back exactly what they requested. Let me know your thoughts.

Reply:
I'm thinking that the `delete` function should automatically do this for you. For example if any of our code is modifying what data is available we should keep the data we return from those functions to be consistent.

Reply:
I see. Another way I can write this is call `syn.get()` on the synID string rather than the Entity object to avoid modifying data between steps.